src/metabase/api/dataset.clj

```diff
@@ -55,10 +55,14 @@
       (events/publish-event! :table-read (assoc (Table table-id) :actor_id api/*current-user-id*))))
   ;; add sensible constraints for results limits on our query
   (let [source-card-id (query->source-card-id query)
-        info           {:executed-by api/*current-user-id*
-                        :context     context
-                        :card-id     source-card-id
-                        :nested?     (boolean source-card-id)}]
+        source-card    (when source-card-id
+                            (Card source-card-id))
```

Comment:
```suggestion
        source-card    (when source-card-id
                         (db/select-one [Card :result_metadata :dataset] :id source-card-id))
```


You're only using those two columns and it's extremely easy to limit the query to just those so might as well do it